### PR TITLE
Fix small regression with allowing a nil path in [NSData dataWithContentsOfFile:]

### PR DESCRIPTION
**Pull request by bbowman**
Fix small regression with allowing a nil path in [NSData dataWithContentsOfFile:]

---
This change is [Reviewable](https://reviewable.io/reviews/microsoft/winobjc/1232)

### CODE REVIEW

**Comment by msft-Jeyaram** [comment]
>     return nil;

[self release]?

---
Refers to: Frameworks/Foundation/NSCFData.mm:62 in 7ac059b. [](commit_id = 7ac059b325ccc141fb5f4b03654c1d6e63ec567c, deletion_comment = False)

**Comment by msft-Jeyaram** [comment]
>     return nil;

[self release]?

---
Refers to: Frameworks/Foundation/NSData.mm:410 in 7ac059b. [](commit_id = 7ac059b325ccc141fb5f4b03654c1d6e63ec567c, deletion_comment = False)

**Comment by msft-Jeyaram** [comment]
>     StrongId<NSString> base64String = [[[NSString alloc] initWithData:base64Data encoding:NSUTF8StringEncoding] autorelease];

this can be invalid.

---
Refers to: Frameworks/Foundation/NSData.mm:417 in 7ac059b. [](commit_id = 7ac059b325ccc141fb5f4b03654c1d6e63ec567c, deletion_comment = False)

**Comment by msft-Jeyaram** [comment]
> - (instancetype)initWithBytes:(const void*)bytes length:(unsigned)length {

check for invalid bytes?

---
Refers to: Frameworks/Foundation/NSData.mm:225 in 7ac059b. [](commit_id = 7ac059b325ccc141fb5f4b03654c1d6e63ec567c, deletion_comment = False)

**Comment by msft-Jeyaram** [comment]
> - (void)getBytes:(void*)dest range:(NSRange)range {

check for invalid dest?

---
Refers to: Frameworks/Foundation/NSData.mm:279 in 7ac059b. [](commit_id = 7ac059b325ccc141fb5f4b03654c1d6e63ec567c, deletion_comment = False)

**Comment by bbowman** [comment]
> - (void)getBytes:(void*)dest range:(NSRange)range {

not part of this change.

---
In reply to: [256453172](https://github.com/Microsoft/WinObjC/pull/1232#issuecomment-256453172) [](ancestors = 256453172)

---
Refers to: Frameworks/Foundation/NSData.mm:279 in 7ac059b. [](commit_id = 7ac059b325ccc141fb5f4b03654c1d6e63ec567c, deletion_comment = False)

**Comment by bbowman** [comment]
> - (instancetype)initWithBytes:(const void*)bytes length:(unsigned)length {

not part of this change

---
In reply to: [256452673](https://github.com/Microsoft/WinObjC/pull/1232#issuecomment-256452673) [](ancestors = 256452673)

---
Refers to: Frameworks/Foundation/NSData.mm:225 in 7ac059b. [](commit_id = 7ac059b325ccc141fb5f4b03654c1d6e63ec567c, deletion_comment = False)

**Comment by msft-Jeyaram** [comment]
> - (instancetype)initWithBytes:(const void*)bytes length:(unsigned)length {

This might cause a bug in the future.
It's a pretty trivial change.

---
In reply to: [256463416](https://github.com/Microsoft/WinObjC/pull/1232#issuecomment-256463416) [](ancestors = 256463416,256452673)

---
Refers to: Frameworks/Foundation/NSData.mm:225 in 7ac059b. [](commit_id = 7ac059b325ccc141fb5f4b03654c1d6e63ec567c, deletion_comment = False)

**Comment by msft-Jeyaram** [comment]
:shipit:

**Comment by MSFTFox** [comment]
:shipit:

**Comment by bbowman** [comment]
> - (instancetype)initWithBytes:(const void*)bytes length:(unsigned)length {

feel free to file a different issue for it. There are a number of places where we are ok crashing in cases of invalid arguments and we'd want to take the time to do the right thing here. It is trivial but as seen with this change, trivial changes may have bigger impacts than first thought (hooray objective C). 

---
In reply to: [256464010](https://github.com/Microsoft/WinObjC/pull/1232#issuecomment-256464010) [](ancestors = 256464010,256463416,256452673)

---
Refers to: Frameworks/Foundation/NSData.mm:225 in 7ac059b. [](commit_id = 7ac059b325ccc141fb5f4b03654c1d6e63ec567c, deletion_comment = False)

**Comment by pradipd** [comment]
:shipit:

**Comment by ms-jihua** [comment]
:shipit:

File: Frameworks/Foundation/NSCFData.mm [hunk 1]
```diff
@@ -46,13 +46,20 @@ - (_Nullable instancetype)initWithBytesNoCopy:(void*)bytes length:(NSUInteger)le
 - (_Nullable instancetype)initWithContentsOfFile:(NSString*)filename options:(NSDataReadingOptions)options error:(NSError**)error {
     CFDataRef tempData;
     SInt32 cfError{};
-    
-    if (CFURLCreateDataAndPropertiesFromResource(nullptr, static_cast<CFURLRef>([NSURL fileURLWithPath:filename]), &tempData, nullptr, nullptr, &cfError)) {
+
+    if (!filename) {
```
Comment (bbowman) [inline Frameworks/Foundation/NSCFData.mm]: I prefer early returns for failure conditions as it gets it out of the way quickly. What you suggest is a pattern that can and does to severely nested code. I agree that this function is kinda cluttered though but meh.

---
In reply to: [85214139](https://github.com/Microsoft/WinObjC/pull/1232#discussion_r85214139) [](ancestors = 85214139)

**Pull request merged**